tokio-postgres/src/connect.rs

```diff
@@ -83,6 +132,433 @@ where
     Err(error.unwrap())
 }
 
+pub async fn yb_connect<T>(
+    mut tls: T,
+    config: &Config,
+) -> Result<(Client, Connection<Socket, T::Stream>), Error>
+where
+    T: MakeTlsConnect<Socket>,
+{
+    if config.host.is_empty() && config.hostaddr.is_empty() {
+        return Err(Error::config("both host and hostaddr are missing".into()));
+    }
+
+    if !config.host.is_empty()
+        && !config.hostaddr.is_empty()
+        && config.host.len() != config.hostaddr.len()
```

Comment:
How are `host` and `hostaddr` different?

Reply:
It is exapled here in the comments: https://github.com/yugabyte/rust-postgres/blob/master/postgres/src/config.rs#L39

Reply:
Discussed on call.